application/src/main/kotlin/net/corda/v5/application/persistence/PersistenceService.kt

```diff
@@ -105,79 +103,62 @@ interface PersistenceService {
-     * @param postProcessorName the name of the post-processor that will process named query results.
-     * @param R the type of the results
-     * @return Cursor configured to poll data for this named query.
+     * @param R the type of the results.
+     * @param limit the maximum number of results to return.
+     * @return list of entities found. Empty list if none were found.
      * @throws CordaPersistenceException if an error happens during query operation
      */
     @Suspendable
     fun <R> query(
         queryName: String,
         namedParameters: Map<String, Any>,
-        postProcessorName: String
-    ): Cursor<R>
+        limit: Int,
+    ): List<R>
 
     /**
-     * Execute a named query in a single transaction and apply post-filtering and post-processing to the results. Casts results to the specified type [R].
+     * Execute a named query in a single transaction, skip results to the specified [offset] and limit the number of returned results. Casts results to the specified type [R].
      *
      * @param queryName the name of the named query registered in the persistence context.
      * @param namedParameters the named parameters to be set in the named query.
-     * @param postFilter the filter to be applied after named query execution.
-     * @param postProcessorName the name of the post-processor that will process named query results.
-     * @param R the type of the results
-     * @return Cursor configured to poll data for this named query.
+     * @param R the type of the results.
+     * @param offset the index of the first result in the query to return.
+     * @param limit the maximum number of results to return.
+     * @return list of entities found. Empty list if none were found.
      * @throws CordaPersistenceException if an error happens during query operation
      */
     @Suspendable
     fun <R> query(
         queryName: String,
         namedParameters: Map<String, Any>,
-        postFilter: NamedQueryFilter,
-        postProcessorName: String
-    ): Cursor<R>
+        offset: Int,
+        limit: Int,
+    ): List<R>
 
     /**
-     * Execute a named query in a single transaction with optional post-filtering and post-processing applied to the results. Casts results to the specified type [R].
+     * Execute a named query in a single transaction with optional offset and limit applied to the results. Casts results to the specified type [R].
      *
      * @param persistenceQueryRequest the request containing information to execute named queries with optional filtering and post-processing
      * @param R the type of the results
-     * @return Cursor configured to poll data for this named query.
+     * @return list of entities found. Empty list if none were found.
      * @throws CordaPersistenceException if an error happens during query operation
      */
     @Suspendable
-    fun <R> query(persistenceQueryRequest: PersistenceQueryRequest): Cursor<R>
```

Comment:
I think we should drop this or keep it depending on what the answer to my comment above is. Since we added the `PersistenceQueryRequest` due to the amount of properties there were at the time (like 5?). The current max is 4 with the offset + limit so the benefit has gone down a bit. Curious what others think.